cartography/intel/azure/storage.py

```diff
@@ -0,0 +1,743 @@
+    s.secondarylocation = account.secondary_location,
+    s.provisioningstate = account.provisioning_state,
+    s.statusofprimary = account.status_of_primary,
+    s.statusofsecondary = account.status_of_secondary,
+    s.supportshttpstrafficonly = account.enable_https_traffic_only
+    WITH s
+    MATCH (owner:AzureSubscription{id: {AZURE_SUBSCRIPTION_ID}})
+    MERGE (owner)-[r:RESOURCE]->(s)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {azure_update_tag}
+    """
+
+    neo4j_session.run(
+        ingest_storage_account,
+        storage_accounts_list=storage_account_list,
+        AZURE_SUBSCRIPTION_ID=subscription_id,
+        azure_update_tag=azure_update_tag,
+    )
+
+
+@timeit
+def sync_storage_account_details(
+        neo4j_session: neo4j.Session, credentials: Credentials, subscription_id: str,
+        storage_account_list: List[Dict], sync_tag: int,
+) -> None:
+    details = get_storage_account_details(credentials, subscription_id, storage_account_list)
+    load_storage_account_details(neo4j_session, credentials, subscription_id, details, sync_tag)
+
+
+@timeit
+def get_storage_account_details(
+        credentials: Credentials, subscription_id: str, storage_account_list: List[Dict],
+) -> Generator[Any, Any, Any]:
+    """
+    Iterates over all Storage Accounts to get the different storage services.
+    """
+    for storage_account in storage_account_list:
+        queue_services = get_queue_services(credentials, subscription_id, storage_account)
+        table_services = get_table_services(credentials, subscription_id, storage_account)
+        file_services = get_file_services(credentials, subscription_id, storage_account)
+        blob_services = get_blob_services(credentials, subscription_id, storage_account)
+        yield storage_account['id'], storage_account['name'], storage_account[
+            'resourceGroup'
+        ], queue_services, table_services, file_services, blob_services
+
+
+@timeit
+def get_queue_services(credentials: Credentials, subscription_id: str, storage_account: Dict) -> List[Dict]:
+    """
+    Gets the list of queue services.
+    """
+    try:
+        client = get_client(credentials, subscription_id)
+        queue_service_list = client.queue_services.list(
+            storage_account['resourceGroup'], storage_account['name'],
+        ).as_dict()['value']
+
+    except HttpResponseError as e:
+        logger.warning(f"Error while retrieving queue services list - {e}")
+        return []
+
+    return queue_service_list
+
+
+@timeit
+def get_table_services(credentials: Credentials, subscription_id: str, storage_account: Dict) -> List[Dict]:
+    """
+    Gets the list of table services.
+    """
+    try:
+        client = get_client(credentials, subscription_id)
+        table_service_list = client.table_services.list(
+            storage_account['resourceGroup'], storage_account['name'],
+        ).as_dict()['value']
+
+    except HttpResponseError as e:
```

Comment:
Just like in #563 , lets catch some specific errors.

Reply:
@ramonpetgrave64 Done! :)